addons/payment/models/payment_acquirer.py

```diff
@@ -224,66 +197,65 @@ def render(self, cr, uid, id, reference, amount, currency_id, partner_id=False,
+        cust_method_name = '%s_form_generate_values' % (self.provider)
         if hasattr(self, cust_method_name):
             method = getattr(self, cust_method_name)
-            values = method(cr, uid, id, values, context=context)
+            inv_type = method(values)
+            values = inv_type[0] if isinstance(inv_type, list) else inv_type
 
         values.update({
-            'tx_url': context.get('tx_url', self.get_form_action_url(cr, uid, id, context=context)),
-            'submit_class': context.get('submit_class', 'btn btn-link'),
-            'submit_txt': context.get('submit_txt'),
-            'acquirer': acquirer,
-            'user': self.pool.get("res.users").browse(cr, uid, uid, context=context),
-            'context': context,
+            'tx_url': self.env.context.get('tx_url', self.get_form_action_url()),
+            'submit_class': self.env.context.get('submit_class', 'btn btn-link'),
+            'submit_txt': self.env.context.get('submit_txt'),
+            'acquirer': self,
+            'user': self.env.user,
+            'context': self.env.context,
             'type': values.get('type') or 'form',
         })
         values.setdefault('return_url', False)
 
         # because render accepts view ids but not qweb -> need to use the xml_id
-        return self.pool['ir.ui.view'].render(cr, uid, acquirer.view_template_id.xml_id, values, engine='ir.qweb', context=context)
-
-    def _registration_render(self, cr, uid, id, partner_id, qweb_context=None, context=None):
-        acquirer = self.browse(cr, uid, id, context=context)
-        if qweb_context is None:
-            qweb_context = {}
-        qweb_context.update(id=id, partner_id=partner_id)
-        method_name = '_%s_registration_form_generate_values' % (acquirer.provider,)
+        return self.env['ir.ui.view']._model.render(self.env.cr, self.env.uid, self.view_template_id.xml_id, values, context=self.env.context)
+
+    def _registration_render(self, partner_id, values={}):
+        values.update(id=self.id, partner_id=partner_id)
+        method_name = '_%s_registration_form_generate_values' % (self.provider)
         if hasattr(self, method_name):
             method = getattr(self, method_name)
-            qweb_context.update(method(cr, uid, id, qweb_context, context=context))
-        return self.pool['ir.ui.view'].render(cr, uid, acquirer.registration_view_template_id.xml_id, qweb_context, engine='ir.qweb', context=context)
-
-    def s2s_process(self, cr, uid, id, data, context=None):
-        acquirer = self.browse(cr, uid, id, context=context)
-        cust_method_name = '%s_s2s_form_process' % (acquirer.provider)
-        if not self.s2s_validate(cr, uid, id, data, context=context):
+            values.update(method(values))
+        return self.env['ir.ui.view']._model.render(self.env.cr, self.env.uid, self.registration_view_template_id.xml_id, values, context=self.env.context)
```

Comment:
Because someone fucked up render method in ir_ui_view by adding a wrong decorator.

The way to do this
1/ view = self.env.ref(my_xml_id)
2/ view.render(...)

Pay attention to raise if not found.